tools/star.wdl

```diff
@@ -18,49 +16,54 @@ task build_star_db {
     }
 
     input {
-        Int ncpu = 1
         File reference_fasta
         File gtf
-        String stardb_dir_name = "star_db"
+        String stardb_name = "star_db"
+        Boolean use_all_cores = false
         Int memory_gb = 50
-        Int? disk_size_gb
+        Int modify_disk_size_gb = 0
+        Int ncpu = 1
         Int max_retries = 1
-        Boolean use_all_cores = false
     }
 
-    String stardb_out_name = stardb_dir_name + ".tar.gz"
+    String stardb_out_name = stardb_name + ".tar.gz"
     Float reference_fasta_size = size(reference_fasta, "GiB")
     Float gtf_size = size(gtf, "GiB")
-    Int disk_size = select_first([disk_size_gb, ceil(((reference_fasta_size + gtf_size) * 3) + 10)])
+    Int disk_size_gb = ceil(
+        ((reference_fasta_size + gtf_size) * 3) + 10
+    ) + modify_disk_size_gb
+
+    String memory_limit_bytes = (memory_gb - 2) + "000000000"
 
     command <<<
         set -euo pipefail
 
         n_cores=~{ncpu}
-        if [ "~{use_all_cores}" = "true" ]; then
+        if ~{use_all_cores}; then
             n_cores=$(nproc)
         fi
 
         orig_gtf=~{gtf}
         gtf_name=$(basename "${orig_gtf%.gz}")
-        gunzip -c ~{gtf} > "$gtf_name" || cp ~{gtf} "$gtf_name"
+        gunzip -c ~{gtf} > "$gtf_name" || ln -s ~{gtf} "$gtf_name"
```

Comment:
Isn't `$(basename "${orig_gtf%.gz}")` equivalent to doing `basename $orig_gtf ".gz"`? 
Also is the `ln -s` (or `cp`) necessary? If the input wasn't gzipped, then `orig_gtf` should be the same as `gtf` right?

Reply:
I suppose this might be localizing the file for STAR. We should add some comments to this command though.

Reply:
> Isn't `$(basename "${orig_gtf%.gz}")` equivalent to doing `basename $orig_gtf ".gz"`?

yes that should be changed to just use basename

> Also is the `ln -s` (or `cp`) necessary? If the input wasn't gzipped, then `orig_gtf` should be the same as `gtf` right?

I think the main purpose of the `ln -s` (and what was a `cp`) is so that the same variable (`$gtf_name`) can be passed to STAR. As opposed to having along the lines of `if (gzipped) then $var1  else $var2`.

I would hope the files don't need to be localized for STAR, though I'm not sure. Regardless, this section can definitely get cleaned up